code/modules/events/undead.dm

```diff
@@ -0,0 +1,62 @@
+
+/datum/round_event_control/undead
+	name = "Raise Dead"
+	typepath = /datum/round_event/ghost_role/undead
+	weight = 10
+
+/datum/round_event/ghost_role/undead
+	minimum_required = 1
+	role_name = "friendly zombie"
+	var/spawns = 3
+	fakeable = TRUE
+
+/datum/round_event/ghost_role/undead/announce(fake)
+	var/source = pick("a powerful spell from a far away wizard", "a strange anomalous enegy", "the influence of a high level necromancer", "the power of love", "the will of an omnipotent God", "\[REDACTED\]")
+	var/purpose = pick("party like there's no tomorrow", "seek employment", "partake in the democratic process", "fulfill their new year's promises", "have an after-life crysis", "preach about their new religion", "answer your questions about life, the universe, and everything", "persuade you not to treat them like their brain eating counterparts", "repair their past mistakes", "\[REDACTED\]")
+
+	priority_announce("Your station has been subjected to [source]. Some dead bodies have come back to life to [purpose]. On behalf of Nanotrasen, please welcome these former employees as your new coworkers!","[command_name()] Medium-Priority Update")
+
+/datum/round_event/ghost_role/undead/spawn_role()
+	var/list/mob/dead/observer/candidates = get_candidates(ROLE_LAVALAND, null, ROLE_LAVALAND)
```

Comment:
Make a new role for this, the candidates is used for the No for the rest of the round option so should be unique for spawns.